source/server/config/network/mongo_proxy.cc

```diff
@@ -5,28 +5,37 @@
 #include "envoy/network/connection.h"
 #include "envoy/registry/registry.h"
 
-#include "common/json/config_schemas.h"
+#include "common/config/filter_json.h"
+#include "common/json/json_loader.h"
 #include "common/mongo/proxy.h"
 
+#include "api/filter/network/mongo_proxy.pb.h"
+
 namespace Envoy {
 namespace Server {
 namespace Configuration {
 
-NetworkFilterFactoryCb
-MongoProxyFilterConfigFactory::createFilterFactory(const Json::Object& config,
-                                                   FactoryContext& context) {
-  config.validateSchema(Json::Schema::MONGO_PROXY_NETWORK_FILTER_SCHEMA);
+NetworkFilterFactoryCb MongoProxyFilterConfigFactory::createMongoProxyFactory(
+    const envoy::api::v2::filter::network::MongoProxy& mongo_proxy, FactoryContext& context) {
+
+  if (mongo_proxy.stat_prefix().empty()) {
+    throw MissingFieldException("stat_prefix", mongo_proxy);
```

Comment:
@htuch should we include this since this would be covered by validators? My thinking is to remove this check and possibly just have an ASSERT.

Reply:
We can't run some negative tests without these. Thats the reason for adding these.

Reply:
But if it's code we don't need why would we add it? I would just put in an assert and remove the tests?